### ISSUE TITLE
Error while processing form response! (class java.lang.Integer cannot be cast to class java.lang.Boolean)

### ISSUE THREAD

**Issue by DxrkCat** [GeyserMC/Geyser#4289]
### Describe the bug

(velocity.jar:3.2.0-SNAPSHOT (git-8dcc7ee3-b294)
adoptium java 21
error: https://paste.gg/p/anonymous/77fd98a18918441bac7c4afbff5b886f/files/0c4f9a0d18bf42c291dac93eb02dce18/raw
I think this is because the wrong value was received in the packet.

### To Reproduce

Go to the server with Velocity that has Geyser-Velocity and floodgate installed.

### Expected behaviour

idk

### Geyser Version

https://github.com/GeyserMC/Geyser/commit/f40ca2004e2828d40df368432399bebd681d3c4b

**Comment by DxrkCat** [comment]
An important addition this does not happen all the time, but only sometimes some players come in they are fine. And some of them have such errors.

**Comment by onebeastchris** [comment]
Please send a screenshot of the settings menu that causes the issue - there's a chance that the logic to parse responses isn't fully working, and seeing which options exist would probably help :)

**Comment by DxrkCat** [comment]
> Please send a screenshot of the settings menu that causes the issue - there's a chance that the logic to parse responses isn't fully working, and seeing which options exist would probably help :)

I don't know it's the players logging into the server, not me and finding out what settings do it is problematic.

**Comment by onebeastchris** [comment]
Could you ask an affected player to send a screenshot?
It's less about which toggle "does it", and just about seeing which ones are shown at all. Alternatively, I could send you a debug build that would help pin down the issue

**Comment by DxrkCat** [comment]
> Could you ask an affected player to send a screenshot?
> It's less about which toggle "does it", and just about seeing which ones are shown at all. Alternatively, I could send you a debug build that would help pin down the issue

let's send the debug build

**Comment by DxrkCat** [comment]
this error happens when I click on confirm to go to hub on anarchy.

I.e., if a player has an open menu and is moved to another server, the client will automatically send a closing packet, which geyser will not process and throw an exception. 

https://paste.gg/p/anonymous/1d79ac1366624439b00fc60777d5bec9

**Comment by DxrkCat** [comment]
Yes definitely my theory is correct it happens when inventory is open and when switching to another server.

**Comment by onebeastchris** [comment]
The second inventory cast issue is a different one; and likely related to https://github.com/GeyserMC/Geyser/issues/4141.
As for the initial issue; does your server change whether players can see coordinates?
If that's a no, i'll link a debug build that would hopefully narrow the issue down.
